pkg/issuer/vault/issue.go

```diff
@@ -130,7 +132,80 @@ func (v *Vault) configureCertPool(cfg *vault.Config) error {
 	return nil
 }
 
+var (
+	clientRepo   map[*v1alpha1.VaultIssuer]*vault.Client
```

Comment:
Using a pointer to the vault issuer instance won't work here, as the VaultIssuer in the underlying kubernetes client may be a different object in memory on subsequent calls to the `initVaultClient` function.

If you look at the implementation of the new ACME client caching, a `repoKey` struct is defined which contains non-ptr keys for the settings used to construct each client:

https://github.com/jetstack/cert-manager/blob/fd4d2022afd68048a8b12e1866c2c45ed9dbb406/pkg/acme/acme.go#L152-L157

We then have a map of *non-ptr* types to the corresponding client 😄 

https://github.com/jetstack/cert-manager/blob/fd4d2022afd68048a8b12e1866c2c45ed9dbb406/pkg/acme/acme.go#L143-L150

Reply:
that's a bumper. I was thinking the pointer would remain the same. The reason I didn't use non-ptr keys such as vault approle or tokensecret parameters because some of these information (for example, authPath for approle) was defined outside of the initVaultClient() function. I will have to think on how to get these keys for the map.